src/index.html

```diff
@@ -1,20 +1,21 @@
-<!DOCTYPE html>
+﻿<!DOCTYPE html>
 <html>
-    <head>
-        <meta charset="utf-8" />
-        <meta name="viewport" content="width=device-width, initial-scale=1">
-        <title>Epic</title>
-        <style>
-            #canvas { border: 1px solid black; width: 100%; height: 80vh }
-            #parameter-slider { width: 50% }
-        </style>
-    </head>
-    <body>
-        <div id="controls">
-            <input type="button" value="Clear" id="clear-button">
-            <input type="range" min="0" max="1000" value="0" id="parameter-slider">
-        </div>
-        <canvas id="canvas"></canvas>
-        <script src="index.ts"></script>
-    </body>
+<head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>Epic</title>
+    <style>
+        #canvas { border: 1px solid black; width: 100%; height: 80vh }
+        #parameter-slider { width: 50% }
+    </style>
+</head>
+<body>
+    <div id="controls">
+        <input type="button" value="Clear" id="clear-button" />
+        <input type="range" min="0" max="1000" value="0" id="parameter-slider" />
+        <input type="number" min="0" max="4096" value="0" id="complexity-number" />
+    </div>
+    <canvas id="canvas"></canvas>
+    <script src="index.ts"></script>
+</body>
```

Comment:
I should add an .editorconfig to avoid this. It looks like you're converting the line endings in this file. Was this intentional? If not, can you do `git add --renormalize .`?

Reply:
Will do that.
I configured Git as "pull end lines as-is, push as-is", I might have copy-pasted the code to my (windows) Notepad.

Reply:
My settings are always on auto.